testing/test_pathlib.py

```diff
@@ -669,6 +707,34 @@ def __init__(self) -> None:
         mod = import_path(init, root=tmp_path, mode=ImportMode.importlib)
         assert len(mod.instance.INSTANCES) == 1
 
+    def test_importlib_doctest(self, monkeypatch: MonkeyPatch, tmp_path: Path) -> None:
```

Comment:
This test was borrowed from #11931 just to ensure the fix proposed in this PR also covers this test. 

However given this use case is also covered by `test_import_using_normal_mechanism_first` so I believe we can drop this test before merging.